naz/client.py

```diff
@@ -1780,6 +1780,8 @@ async def receive_data(self, TESTING: bool = False) -> typing.Union[None, bytes]
                 )
                 # close connection. it will be automatically reconnected later
                 await self._unbind_and_disconnect()
+                await self.re_establish_conn_bind(smpp_command="", log_id="")
+
```

Comment:
@theLastOfCats can I suggest that instead of this change.  

You add a new line after; https://github.com/komuw/naz/blob/3a978348fa7b07653dd224323622c2d467dfceac/naz/client.py#L2362 with 
```python
self.current_session_state = SmppSessionState.CLOSED
```

Reply:
If I'll choose this path, test which introduced in #208 will break.

To fix this flow, we'll need to move the reconnection phase in `send_data`. 
Specifically this part https://github.com/komuw/naz/blob/master/naz/client.py#L1471

Or we reconnect after unbind as in PR.